FSI + kOmegaSST turbulence model

vidulejs opened the pull request:
As we agreed, this is the pull request where we can discuss the issue with the Poisson wall distance calculation and eventually create a new turbulent FSI tutorial. This case would have to be modified to have a higher Re number.

@MakisH 
@KariimAhmed 

<!-- Please submit your Pull Request to the `develop` branch.

It may help to have a look at the file `CONTRIBUTING.md` for a few hints and guidelines. -->

Checklist:

- [ ] I added a summary of any user-facing changes (compared to the last release) in the `changelog-entries/<PRnumber>.md`.
- [ ] I will remember to squash-and-merge, providing a useful summary of the changes of this PR.

vidulejs committed:
added kOmegaSST turbulence model

MakisH committed:
Adjust indentation and remove not needed headers from new files

MakisH committed:
Adjust indentation in turbulenceProperties

MakisH committed:
Cleanup comments in fvSchemes

MakisH committed:
Fix indentation in fvSolution

MakisH committed:
Fix (more) indentation in fvSolution

MakisH commented:
Thanks for pushing this! @KariimAhmed could you please try to adjust this case to reproduce your issue? It would then be much easier for us to fix.

After we have a fix, we should indeed convert this to a new `perpendicular-flap-turbulent` tutorial.

@vidulejs I know this was meant to be a complete draft. I just went ahead changed some formatting things to make this already easier to review and transform into a tutorial, so that you can focus on the actual issue. Please pull from your branch.

vidulejs committed:
Use Poisson wallDist

vidulejs commented:
Thanks for fixing the formatting @MakisH! 

Just to re-iterate from our conversation, we want to create a fully turbulent FSI test case to see if an adapter-related issue appears. We should start with increasing the velocity, to achieve a higher Reynolds number. The numerical schemes will probably need updating, because the turbulent energy dissipation rate `omega` was exploding. 

I noticed I accidentally pushed the wrong `wallDist` method, but during our meeting I had the `Poisson` method and the case was running succesfully.

KariimAhmed commented:
Hello Daniels , Is there any updates regarding the test case ?

vidulejs committed:
increase velocity 10x

vidulejs committed:
fix stability issues with k and omega

vidulejs committed:
refine mesh and extend domain

vidulejs committed:
increase Youngs modulus 50x

vidulejs commented:
Hi Karim!

I've updated the test case and it's working now. I increased the velocity so that the turbulence is more pronounced. K and Omega were unstable so I refined the mesh near the flap and added relaxation factors 0.4. You can see all my changes in the commits.

![image](https://github.com/user-attachments/assets/c5d055c4-c932-4d94-8db5-5e9a58ccaef4)

Let's try to replicate your error!

To start, you could post your latest error logs here. 

We had a guess that the error is related to the `Poisson` `wallDist` method. Currently the test case is running succesfully with this method.  

To replicate your issue, we need to match the test case conditions to your actual case. What do you see is missing from this test case that is in your case? Please suggest changes. You can pull this branch and make edits, but you will not be able to push them to this branch.

MakisH commented:
This PR is only meant to investigate an issue that @KariimAhmed faced. Current status: we cannot reproduce the issue, we need more input.
